File: solutions/cpp/log-levels/1/log_levels.cpp
#include <string>

namespace log_line {
    std::string message(std::string line) {
        int startAt = line.find(" ") + 1;

        return line.substr(startAt);

    }

    std::string log_level(std::string line) {

        int startAt = line.find("[") + 1 ;
        int stopAt = line.find("]") - 1 ;

        return line.substr(startAt, stopAt);
    }

    std::string reformat(std::string line) {
        return message(line) + " " + "(" + log_level(line) + ")";
    }
}

File: solutions/csharp/log-levels/1/LogLevels.cs
static class LogLine
{
    public static string Message(string logLine)
    {
        int start = logLine.IndexOf(' ') + 1;

        return logLine.Substring(start).Trim();
    }

    public static string LogLevel(string logLine)
    {
        int stop = logLine.IndexOf(']');

        return logLine[1..stop].ToLower().Trim();
    }

    public static string Reformat(string logLine)
    {
        return $"{Message(logLine)} ({LogLevel(logLine)})";
    }
}
